python_modules/libraries/dagster-dbt/dagster_dbt/dbt_project.py

```diff
@@ -0,0 +1,154 @@
+import logging
+import os
+import shutil
+from pathlib import Path
+from typing import Optional, Protocol, Sequence, Union, runtime_checkable
+
+from dagster._annotations import experimental
+from dagster._model import DagsterModel
+
+from .errors import DagsterDbtProjectNotFoundError
+
+logger = logging.getLogger("dagster-dbt.artifacts")
+
+
+def using_dagster_dev() -> bool:
+    return bool(os.getenv("DAGSTER_IS_DEV_CLI"))
+
+
+def parse_on_load_opt_in() -> bool:
+    return bool(os.getenv("DAGSTER_DBT_PARSE_PROJECT_ON_LOAD"))
+
+
+@runtime_checkable
+class ManifestInitFn(Protocol):
+    def __call__(self, project: "DbtProject", *, deploying: bool) -> None: ...
+
+
+def dagster_dbt_manifest_init(
```

Comment:
`manifest_init` was the best name i could come up with for this function that gets called at both project init time and deployment prep time.